repos/repos.go

```diff
@@ -0,0 +1,284 @@
+package repos
+
+import (
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/daedaleanai/reqtraq/linepipes"
+)
+
+// A reference to a remote repository that can be one of:
+//   - An HTTP url referencing a remote git repository
+//   - A ssh url referencing a remote git repository
+//   - A git url referencing a remote git repository
+//   - A local repository that will be cloned
+type RemotePath string
```

Comment:
The purpose is to provide type safety wherever possible. There are cases where converting from one to another makes sense, so for that I use casting. For instance, in the example above the conversion from local repo path to remote path would always be valid, but the opposite would not be true and it would be nice to catch any such errors at compile-time.

Reply:
good point, well presented